packages/core/android/src/main/kotlin/com/datadog/reactnative/UiThreadExecutor.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative
+
+import com.facebook.react.bridge.UiThreadUtil
+
+internal interface UiThreadExecutor {
+    fun runOnUiThread(runnable: Runnable)
+
+    object Provider {
```

Comment:
can be just `companion` to make things shorter

```suggestion
    companion object {
```